Sources_v3/StreamChatUI/ChatChannel/MessageComposerSuggestionsViewController.swift

```diff
@@ -5,169 +5,92 @@
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.register(UICollectionViewCell.self, forCellWithReuseIdentifier: "Cell")
+
+        collectionViewHeightObserver = collectionView.observe(
+            \.contentSize,
+            options: [.new],
+            changeHandler: { [weak self] _, change in
+                guard let self = self, let newSize = change.newValue else { return }
+                self.heightConstraint?.constant = newSize.height
+                self.view.layoutIfNeeded()
+            }
+        )
     }
-    
-    // MARK: - Public
-    
-    open func setupAppearance() {
+
+    override public func setUpAppearance() {
         view.backgroundColor = .clear
-        tableView.backgroundColor = .white
-        tableView.estimatedRowHeight = 40
-        tableView.showsVerticalScrollIndicator = false
+        view.layer.masksToBounds = false
+        view.layer.shadowColor = uiConfig.colorPalette.shadow.cgColor
+        view.layer.shadowOffset = .zero
+        view.layer.shadowRadius = 8
+        view.layer.shadowOpacity = 0.85
     }
-    
-    var heightConstraint: NSLayoutConstraint?
-    var widthConstraint: NSLayoutConstraint?
-    var positionXConstraint: NSLayoutConstraint?
-    var positionYConstraint: NSLayoutConstraint?
-    
-    open func setupLayout() {
+
+    override public func setUpLayout() {
         view.translatesAutoresizingMaskIntoConstraints = false
-        tableView.translatesAutoresizingMaskIntoConstraints = false
-        
-        heightConstraint = .init(
-            item: tableView,
-            attribute: .height,
-            relatedBy: .equal,
-            toItem: nil,
-            attribute: .notAnAttribute,
-            multiplier: 1,
-            constant: 300
-        )
-        
-        widthConstraint = .init(
-            item: tableView,
-            attribute: .width,
-            relatedBy: .equal,
-            toItem: nil,
-            attribute: .notAnAttribute,
-            multiplier: 1,
-            constant: UIScreen.main.bounds.width
-        )
-        
-        positionXConstraint = .init(
-            item: tableView,
-            attribute: .centerX,
-            relatedBy: .equal,
-            toItem: view,
-            attribute: .centerX,
-            multiplier: 1,
-            constant: 200
-        )
-        
-        positionYConstraint = .init(
-            item: tableView,
-            attribute: .centerY,
-            relatedBy: .equal,
-            toItem: view,
-            attribute: .centerY,
-            multiplier: 1,
-            constant: 200
-        )
-        
-        let constraints = [heightConstraint, widthConstraint, positionXConstraint, positionYConstraint].compactMap { $0 }
+
+        heightConstraint = collectionView.heightAnchor.constraint(equalToConstant: 0)
+
+        let constraints = [heightConstraint].compactMap { $0 }
+
         NSLayoutConstraint.activate(constraints)
```

Comment:
Some like to show off :trollface: Since you create the constraint on the line above, I'd say `NSLayoutConstraint.activate([heightConstraint!])` is a completely valid approach here.